### ISSUE TITLE
How convert .vcf file into BIMBAM mean genotype file?

### ISSUE THREAD

**Issue by angelaparodymerino** [genetics-statistics/GEMMA#103]
Hi,

I want to run GEMMA including population structure as covariate, and according to the manual I can only do this through the second input format file for GEMMA, which is called BIMBAM file format. I tried many things to get this BIMBAM mean genotype file (from my .vcf file) but without success. Could someone explain more in detail how to convert a .vcf file into BIMBAM mean genotype file?

Thanks in advance,

Cheers,

'Angela

**Comment by pcarbo** [comment]
@angelaparodymerino Use the `--recode` option in plink2; see [here](http://www.cog-genomics.org/plink/1.9/data#recode) for details. The command will be something like `plink --file myped --recode bimbam --out mybimbam`. You should double-check that the resulting file is in the same format as described in the GEMMA manual.

**Comment by pjotrp** [comment]
Since this is such a common thing, we should add a protocol/script for converting VCF to bimbam.

**Comment by ScientificDesi** [comment]
Hi!
I'm trying to convert a VCF file with dosages to the Mean Genotype File required by Gemma
@pcarbo` --recode bimbam` option in `plink` does not create the **Mean Genotype file** but it creates the **Basic genotype file** format explained here: [](http://www.haplotype.org/download/bimbam-manual.pdf) **Page 5** **3.1 Basic genotype file format** 

Does it matter if the input file for Gemma is in this format:
**REF ALT Dosage_ALT** as opposed to **Minor_Allele Common_Allele Dosage_MinorAllele** 
In other words, does it matter if we provide dosage of ALT allele rather than dosage of minor allele as required by Gemma?

**Comment by ScientificDesi** [comment]
Hi!
I'm trying to convert a VCF file with dosages to the Mean Genotype File required by Gemma
@pcarbo` --recode bimbam` option in `plink` does not create the **Mean Genotype file** but it creates the **Basic genotype file** format explained here: http://www.haplotype.org/download/bimbam-manual.pdf **Page 5** **3.1 Basic genotype file format** 

Does it matter if the input file for Gemma is in this format:
**REF ALT Dosage_ALT** as opposed to **Minor_Allele Common_Allele Dosage_MinorAllele** 
In other words, does it matter if we provide dosage of ALT allele rather than dosage of minor allele as required by Gemma?

**Comment by pcarbo** [comment]
@ScientificDesi Thanks for your interesting in GEMMA. Please post questions (other than bug reports and feature requests) to our [Google Group](https://groups.google.com/forum/#!forum/gemma-discussion).

**Comment by Linghan-Jiang** [comment]
@ScientificDesi Have you solved your problems? I am facing the same problem now.

**Comment by pcarbo** [comment]
@Linghan-Jiang You are welcome to send an email to the Google Group with a description of your  problem.

**Comment by prasunanand** [comment]
Hi, 
I hope this helps :) .
With PLINK , I had issues converting to `BIMBAM`

```
Error: --recode bimbam cannot be used with multi-character allele names.
```


I have been successfully able to convert a vcf file using qctool.
```
$ qctool_v2.0-rc9 -g gatk_exome.vcf -ofiletype bimbam_dosage -og abc.txt

Welcome to qctool
(version: 2.0-rc9, revision 95b281d)

(C) 2009-2017 University of Oxford

Opening genotype files                                      : [                 Opening genotype files                                      : [*****************Opening genotype files                                      : [******************************] (1/1,0.0s,67.4/s)
========================================================================

Input SAMPLE file(s):         Output SAMPLE file:             "(n/a)".
Sample exclusion output file:   "(n/a)".

Input GEN file(s):
                                                    (not computed)  "gatk_exome.vcf"
                                         (total 1 sources, number of snps not computed).
                      Number of samples: 2
Output GEN file(s):             "abc.txt"
Output SNP position file(s):    (n/a)
Sample filter:                  .
# of samples in input files:    2.
# of samples after filtering:   2 (0 filtered out).

========================================================================

filetype hint = "bimbam_dosage", guess is "bimbam_dosage".
Processing SNPs                                             :  (175/?,0.1s,1190.1/s)
Total: 175SNPs.
========================================================================

Number of SNPs:
                     -- in input file(s):                 (not computed).
 -- in output file(s):                175

Number of samples in input file(s):   2.

Output GEN files:                     (175    snps)  "abc.txt"
                                      (total 175 snps).
========================================================================


Thank you for using qctool.

prasun@ubuntu:~/Downloads/test$ cat abc.txt 
rs61769198:1:22848972 A C 2 NA
rs2762904:1:33133968 T C NA 2
rs626251:1:60466814 T C NA 2
rs35944236:1:60466840 TA T NA 2
rs2498982:1:62516683 G C NA 2
rs1137101:1:66058513 A G NA 2
rs259358:1:95709821 T C NA 2
rs6698843:1:109806834 C T 2 NA
rs3789604:1:114354942 T G NA 2
rs11341221:1:144873962 CT C 2 NA
rs8480:1:151733335 T G 2 NA
.:1:157516779 G T NA 2
.:1:197479688 C A NA 2
rs200455452:1:228005073 C T NA 2
rs867599:1:228465370 T G NA 2
rs2437150:1:231488524 C T NA 2
rs3841735:1:235505313 A AT 2 NA
rs2256242:1:237951451 A G 2 NA
rs61747073:2:27682309 C T NA 2
rs60197615:2:32667182 G C NA 2
rs10205833:2:37406680 C G NA 2
rs3761701:2:112705157 T C NA 2
rs6731822:2:113513825 T C 2 NA
.:2:152403960 C A NA 2
rs2284934:2:207998800 C T NA 2
rs355058:3:8667896 T C NA 2
.:3:57368861 CT C NA 2
.:3:62478014 C A 2 NA
rs1962893:3:75787726 G A 2 NA
rs141238251:3:75787739 A T 2 NA
rs2625282:3:101484335 G A NA 2
rs6438544:3:119434527 G C NA 2
.:3:119434528 C A NA 2
.:3:121168152 G T 2 NA
.:3:121822564 G T NA 2
rs7645033:3:122437321 T C NA 2
rs9827878:3:134225969 T C NA 2
.:3:139244744 T C NA 2
rs13116684:4:48037926 A G NA 2
.:4:100572396 T C NA 2
rs1051489:5:32400266 A G NA 2
rs140062546:5:64070617 A AAATTATGAC NA 2
rs6888707:5:74442920 A G 2 NA
rs1878878:5:89990324 A G 1 NA
rs34468716:5:102894673 T C 0 2
rs17493851:5:145519821 G A NA 2
rs1132338:5:178139442 T G 2 NA
rs11743261:5:178996435 A G NA 2
rs2671431:6:335175 A T 2 NA
rs6900196:6:12122645 A G NA 2
rs2074469:6:29407970 T C NA 2
rs8337:6:29523676 C G NA 2
rs8180544:6:47649222 T G NA 2
rs4715630:6:56417282 C T NA 2
rs4140446:6:90448092 C T NA 2
rs1799693:6:106992464 A G NA 2
rs9372653:6:121632008 T A NA 2
rs2327891:6:138754323 A G 2 2
rs2242318:6:159188128 A G NA 2
.:6:160962177 G T NA 2
rs1043965:7:6370144 G A NA 1
rs2192828:7:12376811 A C NA 2
.:7:15584409 G T NA 2
.:7:22202052 C A NA 2
.:7:25266626 C A NA 2
rs6945679:7:31617733 T C NA 2
rs630191:7:42976789 A G 2 2
rs1029357:7:92762681 A G 2 NA
rs78273697:7:100551840 T A NA 2
rs75365561:7:100551842 C T NA 2
rs75524285:7:100551843 T C 2 2
rs77242346:7:100551849 A T 2 2
rs2904850:7:100552644 C A 2 0
rs73398800:7:100552788 C T 2 NA
rs73714266:7:100552867 T C 2 2
rs201550522:7:142460335 A G NA 2
rs200973660:7:142460339 G A NA 2
rs10454320:7:151970856 T A 2 NA
rs56850341:7:151970931 G A 2 NA
rs6320:7:154862621 T A 2 NA
.:8:29023297 C A NA 2
rs2297075:9:390512 C T NA 2
.:9:21334694 T C NA 2
.:9:79117556 C A NA 2
rs58077086:9:86465131 T C NA 2
rs17723637:9:109687403 A G 2 NA
rs3118994:9:129595895 T C NA 2
rs1042058:10:30728101 T C 0 2
rs2291429:10:45958881 A G NA 2
rs3998860:10:70405855 A G 2 NA
rs701865:10:95381773 T A NA 2
rs805657:10:105762591 G A 2 NA
.:10:116013432 G T NA 2
rs201760950:10:126691605 C A NA 2
rs200173220:10:126691608 C T NA 2
rs3198936:10:126691631 T A NA 2
rs75794788:10:126691634 T A NA 2
rs429886:11:2929552 A G NA 2
rs1056963:11:11373635 C T NA 2
rs28681529:11:55872657 A C NA 2
rs614397:11:63883985 T C NA 2
rs4379854:11:63885632 A G NA 2
rs514534:11:74676794 G C 2 2
rs45579043:11:112041128 TTTTGTCCTG T NA 2
rs28398895:11:124266912 A G NA 2
.:11:130064131 G A NA 2
rs1051375:12:2788879 G A 2 NA
rs28403159:12:10588581 T C 2 NA
rs73260771:12:11214360 C T 2 NA
.:12:22646247 C A NA 2
rs1054426:12:30823971 A G 0 2
rs7139272:12:53722128 G A 2 NA
rs6581046:12:55820121 C T 2 NA
rs145970310:12:80889165 T TTTG 2 NA
rs10506873:12:82792942 A G NA 2
rs4762144:12:96653501 C T NA 2
.:12:123067501 G T NA 2
rs206075:13:32913055 A G 2 NA
rs142875900:13:46170719 CCCAGATACTCTTCCTCCT C 2 NA
rs201185418:13:103514737 G A NA 2
rs7995379:13:107145463 G C NA 2
rs2234636:14:21469151 T C NA 2
rs10146717:14:21831112 G A NA 2
.:14:45623122 G T NA 2
rs913742:14:57101682 A G NA 2
rs61977411:14:70515779 G A NA 2
rs1991517:14:81610583 G C 2 2
rs17762463:14:88651962 C T NA 2
rs4465542:14:105407798 T C 2 NA
rs1835183:15:22368862 G A 2 NA
rs112105930:15:44184197 C T NA 2
rs4366668:15:65903398 G A NA 2
rs3825858:15:69728949 A G 2 NA
.:16:3646215 C A NA 2
rs34654419:16:11001770 G T NA 2
.:16:14340388 C A 0 2
.:16:56436917 C A NA 2
rs16956174:16:82033612 T C NA 2
.:16:88712511 G T NA 2
rs1716:17:3632836 G A NA 2
.:17:8363486 C A 2 NA
rs6503235:17:9515777 G A NA 2
rs7232237:18:31324934 A G 2 NA
rs3809979:18:56203447 A G 2 2
rs2302220:19:3434413 C G NA 2
.:19:6413578 G T NA 2
rs2738442:19:11221454 T C NA 2
rs547516:19:21492062 G A NA 2
rs34726149:19:22952111 A G 2 NA
rs80283382:19:33444511 T C 2 NA
rs10422056:19:38378539 G C NA 2
rs31727:19:39915637 G A NA 2
rs11083907:19:48389425 A G NA 2
rs366793:19:53085659 C T NA 2
rs112227284:19:53344051 T C 2 NA
rs10426347:19:53912242 G A 2 NA
.:19:55597421 A G NA 2
rs10417533:19:57931553 C T 2 0
rs6074549:20:1356261 G C 2 NA
rs7266938:20:29625877 G A 0 2
rs11152458:20:29628431 C T NA 2
.:20:44839078 A T NA 2
rs1051266:21:46957794 T C NA 2
rs2839228:21:47786817 C G NA 2
rs2073448:22:22550510 T G NA 2
rs5750741:22:22569448 T A 2 NA
rs2015035:22:30771554 T G 2 NA
rs5845253:22:36587845 A ACT NA 2
rs2235318:22:40800544 C T 2 NA
rs133072:22:41075543 A G 2 NA
.:22:43936213 G A NA 2
.:22:43936228 G T NA 2
rs1047054:X:55172537 G A 2 NA
.:X:107911706 T C NA 2
rs2298301:X:134713855 G A 2 NA
```

**Comment by IsaacDiaz026** [comment]
> Hi,
> I hope this helps :) .
> With PLINK , I had issues converting to `BIMBAM`
> 
> ```
> Error: --recode bimbam cannot be used with multi-character allele names.
> ```
> 
> I have been successfully able to convert a vcf file using qctool.
> 
> ```
> $ qctool_v2.0-rc9 -g gatk_exome.vcf -ofiletype bimbam_dosage -og abc.txt
> 
> Welcome to qctool
> (version: 2.0-rc9, revision 95b281d)
> 
> (C) 2009-2017 University of Oxford
> 
> Opening genotype files                                      : [                 Opening genotype files                                      : [*****************Opening genotype files                                      : [******************************] (1/1,0.0s,67.4/s)
> ========================================================================
> 
> Input SAMPLE file(s):         Output SAMPLE file:             "(n/a)".
> Sample exclusion output file:   "(n/a)".
> 
> Input GEN file(s):
>                                                     (not computed)  "gatk_exome.vcf"
>                                          (total 1 sources, number of snps not computed).
>                       Number of samples: 2
> Output GEN file(s):             "abc.txt"
> Output SNP position file(s):    (n/a)
> Sample filter:                  .
> # of samples in input files:    2.
> # of samples after filtering:   2 (0 filtered out).
> 
> ========================================================================
> 
> filetype hint = "bimbam_dosage", guess is "bimbam_dosage".
> Processing SNPs                                             :  (175/?,0.1s,1190.1/s)
> Total: 175SNPs.
> ========================================================================
> 
> Number of SNPs:
>                      -- in input file(s):                 (not computed).
>  -- in output file(s):                175
> 
> Number of samples in input file(s):   2.
> 
> Output GEN files:                     (175    snps)  "abc.txt"
>                                       (total 175 snps).
> ========================================================================
> 
> 
> Thank you for using qctool.
> 
> prasun@ubuntu:~/Downloads/test$ cat abc.txt 
> rs61769198:1:22848972 A C 2 NA
> rs2762904:1:33133968 T C NA 2
> rs626251:1:60466814 T C NA 2
> rs35944236:1:60466840 TA T NA 2
> rs2498982:1:62516683 G C NA 2
> rs1137101:1:66058513 A G NA 2
> rs259358:1:95709821 T C NA 2
> rs6698843:1:109806834 C T 2 NA
> rs3789604:1:114354942 T G NA 2
> rs11341221:1:144873962 CT C 2 NA
> rs8480:1:151733335 T G 2 NA
> .:1:157516779 G T NA 2
> .:1:197479688 C A NA 2
> rs200455452:1:228005073 C T NA 2
> rs867599:1:228465370 T G NA 2
> rs2437150:1:231488524 C T NA 2
> rs3841735:1:235505313 A AT 2 NA
> rs2256242:1:237951451 A G 2 NA
> rs61747073:2:27682309 C T NA 2
> rs60197615:2:32667182 G C NA 2
> rs10205833:2:37406680 C G NA 2
> rs3761701:2:112705157 T C NA 2
> rs6731822:2:113513825 T C 2 NA
> .:2:152403960 C A NA 2
> rs2284934:2:207998800 C T NA 2
> rs355058:3:8667896 T C NA 2
> .:3:57368861 CT C NA 2
> .:3:62478014 C A 2 NA
> rs1962893:3:75787726 G A 2 NA
> rs141238251:3:75787739 A T 2 NA
> rs2625282:3:101484335 G A NA 2
> rs6438544:3:119434527 G C NA 2
> .:3:119434528 C A NA 2
> .:3:121168152 G T 2 NA
> .:3:121822564 G T NA 2
> rs7645033:3:122437321 T C NA 2
> rs9827878:3:134225969 T C NA 2
> .:3:139244744 T C NA 2
> rs13116684:4:48037926 A G NA 2
> .:4:100572396 T C NA 2
> rs1051489:5:32400266 A G NA 2
> rs140062546:5:64070617 A AAATTATGAC NA 2
> rs6888707:5:74442920 A G 2 NA
> rs1878878:5:89990324 A G 1 NA
> rs34468716:5:102894673 T C 0 2
> rs17493851:5:145519821 G A NA 2
> rs1132338:5:178139442 T G 2 NA
> rs11743261:5:178996435 A G NA 2
> rs2671431:6:335175 A T 2 NA
> rs6900196:6:12122645 A G NA 2
> rs2074469:6:29407970 T C NA 2
> rs8337:6:29523676 C G NA 2
> rs8180544:6:47649222 T G NA 2
> rs4715630:6:56417282 C T NA 2
> rs4140446:6:90448092 C T NA 2
> rs1799693:6:106992464 A G NA 2
> rs9372653:6:121632008 T A NA 2
> rs2327891:6:138754323 A G 2 2
> rs2242318:6:159188128 A G NA 2
> .:6:160962177 G T NA 2
> rs1043965:7:6370144 G A NA 1
> rs2192828:7:12376811 A C NA 2
> .:7:15584409 G T NA 2
> .:7:22202052 C A NA 2
> .:7:25266626 C A NA 2
> rs6945679:7:31617733 T C NA 2
> rs630191:7:42976789 A G 2 2
> rs1029357:7:92762681 A G 2 NA
> rs78273697:7:100551840 T A NA 2
> rs75365561:7:100551842 C T NA 2
> rs75524285:7:100551843 T C 2 2
> rs77242346:7:100551849 A T 2 2
> rs2904850:7:100552644 C A 2 0
> rs73398800:7:100552788 C T 2 NA
> rs73714266:7:100552867 T C 2 2
> rs201550522:7:142460335 A G NA 2
> rs200973660:7:142460339 G A NA 2
> rs10454320:7:151970856 T A 2 NA
> rs56850341:7:151970931 G A 2 NA
> rs6320:7:154862621 T A 2 NA
> .:8:29023297 C A NA 2
> rs2297075:9:390512 C T NA 2
> .:9:21334694 T C NA 2
> .:9:79117556 C A NA 2
> rs58077086:9:86465131 T C NA 2
> rs17723637:9:109687403 A G 2 NA
> rs3118994:9:129595895 T C NA 2
> rs1042058:10:30728101 T C 0 2
> rs2291429:10:45958881 A G NA 2
> rs3998860:10:70405855 A G 2 NA
> rs701865:10:95381773 T A NA 2
> rs805657:10:105762591 G A 2 NA
> .:10:116013432 G T NA 2
> rs201760950:10:126691605 C A NA 2
> rs200173220:10:126691608 C T NA 2
> rs3198936:10:126691631 T A NA 2
> rs75794788:10:126691634 T A NA 2
> rs429886:11:2929552 A G NA 2
> rs1056963:11:11373635 C T NA 2
> rs28681529:11:55872657 A C NA 2
> rs614397:11:63883985 T C NA 2
> rs4379854:11:63885632 A G NA 2
> rs514534:11:74676794 G C 2 2
> rs45579043:11:112041128 TTTTGTCCTG T NA 2
> rs28398895:11:124266912 A G NA 2
> .:11:130064131 G A NA 2
> rs1051375:12:2788879 G A 2 NA
> rs28403159:12:10588581 T C 2 NA
> rs73260771:12:11214360 C T 2 NA
> .:12:22646247 C A NA 2
> rs1054426:12:30823971 A G 0 2
> rs7139272:12:53722128 G A 2 NA
> rs6581046:12:55820121 C T 2 NA
> rs145970310:12:80889165 T TTTG 2 NA
> rs10506873:12:82792942 A G NA 2
> rs4762144:12:96653501 C T NA 2
> .:12:123067501 G T NA 2
> rs206075:13:32913055 A G 2 NA
> rs142875900:13:46170719 CCCAGATACTCTTCCTCCT C 2 NA
> rs201185418:13:103514737 G A NA 2
> rs7995379:13:107145463 G C NA 2
> rs2234636:14:21469151 T C NA 2
> rs10146717:14:21831112 G A NA 2
> .:14:45623122 G T NA 2
> rs913742:14:57101682 A G NA 2
> rs61977411:14:70515779 G A NA 2
> rs1991517:14:81610583 G C 2 2
> rs17762463:14:88651962 C T NA 2
> rs4465542:14:105407798 T C 2 NA
> rs1835183:15:22368862 G A 2 NA
> rs112105930:15:44184197 C T NA 2
> rs4366668:15:65903398 G A NA 2
> rs3825858:15:69728949 A G 2 NA
> .:16:3646215 C A NA 2
> rs34654419:16:11001770 G T NA 2
> .:16:14340388 C A 0 2
> .:16:56436917 C A NA 2
> rs16956174:16:82033612 T C NA 2
> .:16:88712511 G T NA 2
> rs1716:17:3632836 G A NA 2
> .:17:8363486 C A 2 NA
> rs6503235:17:9515777 G A NA 2
> rs7232237:18:31324934 A G 2 NA
> rs3809979:18:56203447 A G 2 2
> rs2302220:19:3434413 C G NA 2
> .:19:6413578 G T NA 2
> rs2738442:19:11221454 T C NA 2
> rs547516:19:21492062 G A NA 2
> rs34726149:19:22952111 A G 2 NA
> rs80283382:19:33444511 T C 2 NA
> rs10422056:19:38378539 G C NA 2
> rs31727:19:39915637 G A NA 2
> rs11083907:19:48389425 A G NA 2
> rs366793:19:53085659 C T NA 2
> rs112227284:19:53344051 T C 2 NA
> rs10426347:19:53912242 G A 2 NA
> .:19:55597421 A G NA 2
> rs10417533:19:57931553 C T 2 0
> rs6074549:20:1356261 G C 2 NA
> rs7266938:20:29625877 G A 0 2
> rs11152458:20:29628431 C T NA 2
> .:20:44839078 A T NA 2
> rs1051266:21:46957794 T C NA 2
> rs2839228:21:47786817 C G NA 2
> rs2073448:22:22550510 T G NA 2
> rs5750741:22:22569448 T A 2 NA
> rs2015035:22:30771554 T G 2 NA
> rs5845253:22:36587845 A ACT NA 2
> rs2235318:22:40800544 C T 2 NA
> rs133072:22:41075543 A G 2 NA
> .:22:43936213 G A NA 2
> .:22:43936228 G T NA 2
> rs1047054:X:55172537 G A 2 NA
> .:X:107911706 T C NA 2
> rs2298301:X:134713855 G A 2 NA
> ```

What did your vcf file look like? I tried this same command in qctool but it doesn't like the formatting of my vcf

**Comment by Noemi-Nicole** [comment]
> > Hi,
> > I hope this helps :) .
> > With PLINK , I had issues converting to `BIMBAM`
> > ```
> > Error: --recode bimbam cannot be used with multi-character allele names.
> > ```
> > 
> > 
> > I have been successfully able to convert a vcf file using qctool.
> > ```
> > $ qctool_v2.0-rc9 -g gatk_exome.vcf -ofiletype bimbam_dosage -og abc.txt
> > 
> > Welcome to qctool
> > (version: 2.0-rc9, revision 95b281d)
> > 
> > (C) 2009-2017 University of Oxford
> > 
> > Opening genotype files                                      : [                 Opening genotype files                                      : [*****************Opening genotype files                                      : [******************************] (1/1,0.0s,67.4/s)
> > ========================================================================
> > 
> > Input SAMPLE file(s):         Output SAMPLE file:             "(n/a)".
> > Sample exclusion output file:   "(n/a)".
> > 
> > Input GEN file(s):
> >                                                     (not computed)  "gatk_exome.vcf"
> >                                          (total 1 sources, number of snps not computed).
> >                       Number of samples: 2
> > Output GEN file(s):             "abc.txt"
> > Output SNP position file(s):    (n/a)
> > Sample filter:                  .
> > # of samples in input files:    2.
> > # of samples after filtering:   2 (0 filtered out).
> > 
> > ========================================================================
> > 
> > filetype hint = "bimbam_dosage", guess is "bimbam_dosage".
> > Processing SNPs                                             :  (175/?,0.1s,1190.1/s)
> > Total: 175SNPs.
> > ========================================================================
> > 
> > Number of SNPs:
> >                      -- in input file(s):                 (not computed).
> >  -- in output file(s):                175
> > 
> > Number of samples in input file(s):   2.
> > 
> > Output GEN files:                     (175    snps)  "abc.txt"
> >                                       (total 175 snps).
> > ========================================================================
> > 
> > 
> > Thank you for using qctool.
> > 
> > prasun@ubuntu:~/Downloads/test$ cat abc.txt 
> > rs61769198:1:22848972 A C 2 NA
> > rs2762904:1:33133968 T C NA 2
> > rs626251:1:60466814 T C NA 2
> > rs35944236:1:60466840 TA T NA 2
> > rs2498982:1:62516683 G C NA 2
> > rs1137101:1:66058513 A G NA 2
> > rs259358:1:95709821 T C NA 2
> > rs6698843:1:109806834 C T 2 NA
> > rs3789604:1:114354942 T G NA 2
> > rs11341221:1:144873962 CT C 2 NA
> > rs8480:1:151733335 T G 2 NA
> > .:1:157516779 G T NA 2
> > .:1:197479688 C A NA 2
> > rs200455452:1:228005073 C T NA 2
> > rs867599:1:228465370 T G NA 2
> > rs2437150:1:231488524 C T NA 2
> > rs3841735:1:235505313 A AT 2 NA
> > rs2256242:1:237951451 A G 2 NA
> > rs61747073:2:27682309 C T NA 2
> > rs60197615:2:32667182 G C NA 2
> > rs10205833:2:37406680 C G NA 2
> > rs3761701:2:112705157 T C NA 2
> > rs6731822:2:113513825 T C 2 NA
> > .:2:152403960 C A NA 2
> > rs2284934:2:207998800 C T NA 2
> > rs355058:3:8667896 T C NA 2
> > .:3:57368861 CT C NA 2
> > .:3:62478014 C A 2 NA
> > rs1962893:3:75787726 G A 2 NA
> > rs141238251:3:75787739 A T 2 NA
> > rs2625282:3:101484335 G A NA 2
> > rs6438544:3:119434527 G C NA 2
> > .:3:119434528 C A NA 2
> > .:3:121168152 G T 2 NA
> > .:3:121822564 G T NA 2
> > rs7645033:3:122437321 T C NA 2
> > rs9827878:3:134225969 T C NA 2
> > .:3:139244744 T C NA 2
> > rs13116684:4:48037926 A G NA 2
> > .:4:100572396 T C NA 2
> > rs1051489:5:32400266 A G NA 2
> > rs140062546:5:64070617 A AAATTATGAC NA 2
> > rs6888707:5:74442920 A G 2 NA
> > rs1878878:5:89990324 A G 1 NA
> > rs34468716:5:102894673 T C 0 2
> > rs17493851:5:145519821 G A NA 2
> > rs1132338:5:178139442 T G 2 NA
> > rs11743261:5:178996435 A G NA 2
> > rs2671431:6:335175 A T 2 NA
> > rs6900196:6:12122645 A G NA 2
> > rs2074469:6:29407970 T C NA 2
> > rs8337:6:29523676 C G NA 2
> > rs8180544:6:47649222 T G NA 2
> > rs4715630:6:56417282 C T NA 2
> > rs4140446:6:90448092 C T NA 2
> > rs1799693:6:106992464 A G NA 2
> > rs9372653:6:121632008 T A NA 2
> > rs2327891:6:138754323 A G 2 2
> > rs2242318:6:159188128 A G NA 2
> > .:6:160962177 G T NA 2
> > rs1043965:7:6370144 G A NA 1
> > rs2192828:7:12376811 A C NA 2
> > .:7:15584409 G T NA 2
> > .:7:22202052 C A NA 2
> > .:7:25266626 C A NA 2
> > rs6945679:7:31617733 T C NA 2
> > rs630191:7:42976789 A G 2 2
> > rs1029357:7:92762681 A G 2 NA
> > rs78273697:7:100551840 T A NA 2
> > rs75365561:7:100551842 C T NA 2
> > rs75524285:7:100551843 T C 2 2
> > rs77242346:7:100551849 A T 2 2
> > rs2904850:7:100552644 C A 2 0
> > rs73398800:7:100552788 C T 2 NA
> > rs73714266:7:100552867 T C 2 2
> > rs201550522:7:142460335 A G NA 2
> > rs200973660:7:142460339 G A NA 2
> > rs10454320:7:151970856 T A 2 NA
> > rs56850341:7:151970931 G A 2 NA
> > rs6320:7:154862621 T A 2 NA
> > .:8:29023297 C A NA 2
> > rs2297075:9:390512 C T NA 2
> > .:9:21334694 T C NA 2
> > .:9:79117556 C A NA 2
> > rs58077086:9:86465131 T C NA 2
> > rs17723637:9:109687403 A G 2 NA
> > rs3118994:9:129595895 T C NA 2
> > rs1042058:10:30728101 T C 0 2
> > rs2291429:10:45958881 A G NA 2
> > rs3998860:10:70405855 A G 2 NA
> > rs701865:10:95381773 T A NA 2
> > rs805657:10:105762591 G A 2 NA
> > .:10:116013432 G T NA 2
> > rs201760950:10:126691605 C A NA 2
> > rs200173220:10:126691608 C T NA 2
> > rs3198936:10:126691631 T A NA 2
> > rs75794788:10:126691634 T A NA 2
> > rs429886:11:2929552 A G NA 2
> > rs1056963:11:11373635 C T NA 2
> > rs28681529:11:55872657 A C NA 2
> > rs614397:11:63883985 T C NA 2
> > rs4379854:11:63885632 A G NA 2
> > rs514534:11:74676794 G C 2 2
> > rs45579043:11:112041128 TTTTGTCCTG T NA 2
> > rs28398895:11:124266912 A G NA 2
> > .:11:130064131 G A NA 2
> > rs1051375:12:2788879 G A 2 NA
> > rs28403159:12:10588581 T C 2 NA
> > rs73260771:12:11214360 C T 2 NA
> > .:12:22646247 C A NA 2
> > rs1054426:12:30823971 A G 0 2
> > rs7139272:12:53722128 G A 2 NA
> > rs6581046:12:55820121 C T 2 NA
> > rs145970310:12:80889165 T TTTG 2 NA
> > rs10506873:12:82792942 A G NA 2
> > rs4762144:12:96653501 C T NA 2
> > .:12:123067501 G T NA 2
> > rs206075:13:32913055 A G 2 NA
> > rs142875900:13:46170719 CCCAGATACTCTTCCTCCT C 2 NA
> > rs201185418:13:103514737 G A NA 2
> > rs7995379:13:107145463 G C NA 2
> > rs2234636:14:21469151 T C NA 2
> > rs10146717:14:21831112 G A NA 2
> > .:14:45623122 G T NA 2
> > rs913742:14:57101682 A G NA 2
> > rs61977411:14:70515779 G A NA 2
> > rs1991517:14:81610583 G C 2 2
> > rs17762463:14:88651962 C T NA 2
> > rs4465542:14:105407798 T C 2 NA
> > rs1835183:15:22368862 G A 2 NA
> > rs112105930:15:44184197 C T NA 2
> > rs4366668:15:65903398 G A NA 2
> > rs3825858:15:69728949 A G 2 NA
> > .:16:3646215 C A NA 2
> > rs34654419:16:11001770 G T NA 2
> > .:16:14340388 C A 0 2
> > .:16:56436917 C A NA 2
> > rs16956174:16:82033612 T C NA 2
> > .:16:88712511 G T NA 2
> > rs1716:17:3632836 G A NA 2
> > .:17:8363486 C A 2 NA
> > rs6503235:17:9515777 G A NA 2
> > rs7232237:18:31324934 A G 2 NA
> > rs3809979:18:56203447 A G 2 2
> > rs2302220:19:3434413 C G NA 2
> > .:19:6413578 G T NA 2
> > rs2738442:19:11221454 T C NA 2
> > rs547516:19:21492062 G A NA 2
> > rs34726149:19:22952111 A G 2 NA
> > rs80283382:19:33444511 T C 2 NA
> > rs10422056:19:38378539 G C NA 2
> > rs31727:19:39915637 G A NA 2
> > rs11083907:19:48389425 A G NA 2
> > rs366793:19:53085659 C T NA 2
> > rs112227284:19:53344051 T C 2 NA
> > rs10426347:19:53912242 G A 2 NA
> > .:19:55597421 A G NA 2
> > rs10417533:19:57931553 C T 2 0
> > rs6074549:20:1356261 G C 2 NA
> > rs7266938:20:29625877 G A 0 2
> > rs11152458:20:29628431 C T NA 2
> > .:20:44839078 A T NA 2
> > rs1051266:21:46957794 T C NA 2
> > rs2839228:21:47786817 C G NA 2
> > rs2073448:22:22550510 T G NA 2
> > rs5750741:22:22569448 T A 2 NA
> > rs2015035:22:30771554 T G 2 NA
> > rs5845253:22:36587845 A ACT NA 2
> > rs2235318:22:40800544 C T 2 NA
> > rs133072:22:41075543 A G 2 NA
> > .:22:43936213 G A NA 2
> > .:22:43936228 G T NA 2
> > rs1047054:X:55172537 G A 2 NA
> > .:X:107911706 T C NA 2
> > rs2298301:X:134713855 G A 2 NA
> > ```
> 
> What did your vcf file look like? I tried this same command in qctool but it doesn't like the formatting of my vcf

I am having your same problem, because qctool works only with vcf file of format v4.2 and not v4.4. Do you know how I can convert my vcf from v4.2 to v4.4 ?
Thanks in advance,
Noemi

**Comment by pjotrp** [comment]
That is cool. You can add this to the docs if you want. I'll merge
that.

**Comment by Noemi-Nicole** [comment]
Thank you but I cannot share my data. Do you know how I can fix the problem ? Because my .vcf files are on v4.2 format actually

**Comment by pjotrp** [comment]
The format of the VCF file is not expected. VCF is pretty lossy as a standard.

**Comment by Noemi-Nicole** [comment]
I apologize for answering with such a delay. It was an issue with the qctool version. 
With qctool_v2.0.1 it works. 
Thank you anyway !
